server/server.go

```diff
@@ -1820,6 +1822,7 @@ func (s *Server) createClient(conn net.Conn) *client {
 	s.mu.Lock()
 	info := s.copyInfo()
 	c.nonce = []byte(info.Nonce)
+	c.headers = info.Headers
```

Comment:
No need to set this here, really it's processConnect connect that needs to set it.

Reply:
Yes agree, as I start to write those next tests will be cleaned up.

Reply:
Actually this is for client to know if server supports headers.. The one sent from the client will show up in clientOpts.

Reply:
I see. 
I would imagine as single flag being more handy going forward.
You support headers for a connection only if (client & server)
Even for the time between connect and processing the connect message we have to assume the client does not support header.